test/function_test.rb

```diff
@@ -35,7 +35,7 @@ def test_throwing_function
       context.call('f', 1)
     end
     assert_equal err.message, 'Error: foo bar'
-    assert_match(/1:23/, err.backtrace[0])
+    assert_match(/1:23/, err.backtrace[0]) unless RUBY_ENGINE == "truffleruby"
```

Comment:
Should try to fix by adding `Polyglot::InnerContext#eval_or_exception`, and then using Truffle::Interop methods to format the backtrace as needed + Polyglot::ExceptionTrait to implement `message` and a default `backtrace`